src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Web/Debugger/FlavorServerSettings.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            var settings = new FlavorServerSettings();
+
+            var webPropertiesRoot = flavorRoot.Element(PropertyNames.RootNode);
+            if (webPropertiesRoot == null)
+            {
+                return settings;
+            }
+
+            if (TryReadBoolValue(webPropertiesRoot, PropertyNames.UsingIIS, out bool boolVal))
+            {
+                settings.UseIIS = boolVal;
+            }
+
+            if (TryReadIntValue(webPropertiesRoot, PropertyNames.ServerPort, out int intVal))
+            {
+                settings.DevelopmentServerPort = intVal;
+            }                
+
+            if (TryReadStringValue(webPropertiesRoot, PropertyNames.ServerVPath, out string? strVal) && strVal != null)
+            {
+                settings.DevelopmentServerVPath = strVal;
+            }    
+            
+            if (TryReadStringValue(webPropertiesRoot, PropertyNames.IISUrl, out strVal))
+            {
+                settings.IISUrl = strVal;
+            }
+            
+            if (TryReadStringValue(webPropertiesRoot, PropertyNames.IISAppRootUrl, out strVal))
+            {
+                settings.IISAppRootOverrideUrl = strVal;
+            }                
+
+            if (TryReadBoolValue(webPropertiesRoot, PropertyNames.OverrideIISAppRootUrl, out boolVal))
+            {
+                settings.UseIISAppRootOverrideUrl = boolVal;
+            }
+            
+            if (TryReadBoolValue(webPropertiesRoot, PropertyNames.UseCustomServer, out boolVal))
+            {
+                settings.UseCustomServer = boolVal;
+            }
+            
+            if (TryReadStringValue(webPropertiesRoot, PropertyNames.CustomServerUrl, out strVal))
+            {
+                settings.CustomServerUrl = strVal;
+            }                
+
+            return settings;
+        }
+
+        public XElement ToXml()
+        {
+            if (UseIIS)
+            {
+            }
+
+            return new XElement(PropertyNames.RootNode);
+        }
+
+        public bool UseIIS { get; private set; }
+        public int SslPort { get; private set; }
+        public int DevelopmentServerPort { get; private set; }
+        public string DevelopmentServerVPath { get; private set; } = "/";
+        public string? IISUrl { get; private set; }
+        public string? IISAppRootOverrideUrl { get; private set; }
+        public bool UseIISAppRootOverrideUrl { get; private set; }
+        public bool UseCustomServer { get; private set; }
+        public string? CustomServerUrl { get; private set; }
+
+        private static bool TryReadBoolValue(XElement root, string propertyName, out bool boolValue)
+        {
+            boolValue = false;
+            string? elementValue = root.Element(propertyName)?.Value;
+            if (!string.IsNullOrEmpty(elementValue)&& bool.TryParse(elementValue, out bool value))
```

Comment:
>!string.IsNullOrEmpty(elementValue)&& [](start = 16, length = 37)

could probably be simplified since the TryParse methods already check null/empty